hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1263,6 +1279,42 @@ bool MDMParser::powerState(void) const {
     return state;
 }
 
+bool MDMParser::powerOnOffGracefully(void) {
```

Comment:
Nit: the naming is a bit confusing. Let's use `softPowerOff()` or something like that similar to what we use on Gen 3.